ledger/simulation/simulation_eval_test.go

```diff
@@ -2080,6 +2081,64 @@ func TestMaxDepthAppWithPCTrace(t *testing.T) {
 	})
 }
 
+func goValuesToTealValues(goValues ...interface{}) []basics.TealValue {
+	if len(goValues) == 0 {
+		return nil
+	}
+
+	boolToUint64 := func(b bool) uint64 {
+		if b {
+			return 1
+		}
+		return 0
+	}
+
+	modelValues := make([]basics.TealValue, len(goValues))
+	for i, goValue := range goValues {
+		switch goValue.(type) {
+		case []byte:
+			convertedBytes, _ := goValue.([]byte)
+			modelValues[i] = basics.TealValue{
+				Type:  basics.TealBytesType,
+				Bytes: string(convertedBytes),
+			}
```

Comment:
nit: if you assign to a variable in the switch, you don't need to cast the value inside of each case, e.g.
```suggestion
		switch convertedValue := goValue.(type) {
		case []byte:
			modelValues[i] = basics.TealValue{
				Type:  basics.TealBytesType,
				Bytes: string(convertedValue),
			}
```